### PR TITLE
change some mods from expiry -1

### PR DESCRIPTION
**Pull request by k0l11**
working on #203 

- done
	- crimson witch 4pc -> 2pc increase (also adjust mod name: cw-4pc -> crimson-4pc-stacks)
	- flower of paradise lost 4pc increase
	- mika a1
	- razor q speed burst (also adjust mod name: speed-burst -> razor-q)
	- razor e energy sigil (also adjust mod name: er-sigil/razor-sigil-duration -> razor-sigil)
	- mappa mare (also adjust mod name: mappa/mappa-mare-stacks -> mappa-mare)
	- beacon
- TODO (part of separate PR after mods and stats array rewrite since these require new enemy mods, namely ones to influence dmg bonus/crit rate buffs tied to an enemy):
	- ganyu c4 damage bonus
	- mona burst damage bonus
	- mona c4 crit rate
	- toukabou shigure damage bonus
- considered changing these but I don't think it's feasible (**person reviewing the PR should double-check**):
	- husk 4pc
	- nymph's dream 4pc
	- aloy coil mod
	- ayaka c6
	- ayato c2
	- candace a4
	- candace burst init
	- heizou c6
	- sayu c2
	- shenhe c4
	- polar star
	- thundering pulse
	- fruit of fulfillment 
	- serpent spine 
	- alley flash 
	- mistsplitter 
	- lost prayer

### CODE REVIEW

File: internal/weapons/catalyst/mappa/mappa.go [hunk 1]
```diff
@@ -40,46 +42,39 @@ func NewWeapon(c *core.Core, char *character.CharWrapper, p weapon.WeaponProfile
...
 		}
+
+		m := make([]float64, attributes.EndStatType)
+		m[attributes.PyroP] = dmg * float64(w.stacks)
+		m[attributes.HydroP] = dmg * float64(w.stacks)
+		m[attributes.CryoP] = dmg * float64(w.stacks)
+		m[attributes.ElectroP] = dmg * float64(w.stacks)
+		m[attributes.AnemoP] = dmg * float64(w.stacks)
+		m[attributes.GeoP] = dmg * float64(w.stacks)
+		m[attributes.DendroP] = dmg * float64(w.stacks)
```
Comment (unleashurgeek) [inline internal/weapons/catalyst/mappa/mappa.go:59]: nit: cleaner to make list of attributes to set and loop over:
```go
for _, attr := range elementAttributes {
  m[attr] = dmg * float64(w.stacks)
}

**Review by unleashurgeek — COMMENTED** [review]
Currently going through the list of other mods in description. Will comment thoughts on each as I go through

**Comment by unleashurgeek** [comment]
husk 4pc: No clean solution imo. Only way I can think of:
  - create a stat mod per stack (husk-4pc-stack-#)
  - create an "add stack" helper function. When adding a stack you reset the durations of all stacks
  - durations should be set such that 1 stack is lost every 6s. IE:
    - `husk-4pc-stack-1`: 6s
    - `husk-4pc-stack-2`: 12s
    - `husk-4pc-stack-3`: 18s
    - `husk-4pc-stack-4`: 24s
  - delete stack loss logic since it now happens naturally with the above durations

**Comment by unleashurgeek** [comment]
nymph's dream 4pc (to me is looks like current implementation is incorrect? In the current state, you'll get a status effect even when active stacks > 3. So then when 1 status effect expires, another will immediately take over without the character doing anything):
* stat mod per type
* in damage event, loop over all possible stat mods. If 3 are active, do nothing
* else, add a new stat mod of that attack type

File: internal/weapons/catalyst/mappa/mappa.go (hunk 1 above)
Comment (k0l11) [reply internal/weapons/catalyst/mappa/mappa.go:59]: leaving as is (many other places do it like this and I don't feel like changing them in this PR)

**Comment by k0l11** [comment]
> to me is looks like current implementation is incorrect? In the current state, you'll get a status effect even when active stacks > 3.
implementation should be correct, see the [artifact description](https://genshin-impact.fandom.com/wiki/Nymph%27s_Dream)

**Comment by k0l11** [comment]
> husk 4pc: No clean solution imo.
> nymph's dream 4pc

fun fact: they also have it as an expiry -1 mod

**Comment by unleashurgeek** [comment]
candace burst init:
* Add attack mod to all chars when burst happens with the given duration
* keep the conditional logic in the amount func

**Comment by unleashurgeek** [comment]
shenhe C4:
* keep the conditional logic in the amount func
* Replace C4BuffKey with adding this mod w/ the 60s duration (and deleting in reset CB)

**Review by unleashurgeek — APPROVED** [review]

**Pull request merged**